lib/trivial.nix

```diff
@@ -53,6 +53,41 @@ rec {
   # argument, but it's nice this way if several uses of `extends` are cascaded.
   extends = f: rattrs: self: let super = rattrs self; in super // f self super;
 
+  # Compose two extension functions:
+  #
+  #     nix-repl> h = self: super: { bar = "<${super.bar}>"; }
+  #
+  #     nix-repl> fix (extends (composeExtensions h g) f)
+  #     { bar = "<bar>"; foo = "foo + "; foobar = "foo + <bar>"; }
+  composeExtensions = f: g: self: super:
```

Comment:
Ah yes. That said, it appears you no longer use this function? `f: g: rattrs: (extend f (extend g rattrs)` seems to be adequate for the situations where I imagine this would be used, even if it is not exactly the same.